File: server/erp-domain/project-service/src/main/java/com/metawebthree/project/domain/entity/Project.java
package com.metawebthree.project.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("pm_project")
public class Project {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String projectCode;

    private String projectName;

    private String description;

    private String status;

    private Long departmentId;

    private String departmentName;

    private Long managerId;

    private String managerName;

    private LocalDate startDate;

    private LocalDate endDate;

    private BigDecimal budgetAmount;

    private BigDecimal usedAmount;

    private String currency;

    private Integer progress;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private String creatorName;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    private String remark;
}

File: apps/backstage-admin/src/apis/project.ts
import http from '@/utils/http'
import type { CommonResult } from '@/types/common'

export interface Project {
  id: number
  projectCode: string
  projectName: string
  description: string
  status: string
  departmentId: number
  departmentName: string
  managerId: number
  managerName: string
  startDate: string
  endDate: string
  budgetAmount: number
  usedAmount: number
  currency: string
  progress: number
  createdBy: number
  creatorName: string
  createdAt: string
  updatedAt: string
  remark: string
}

export interface ProjectCreateCommand {
  projectCode: string
  projectName: string
  description: string
  departmentId: number
  departmentName: string
  managerId: number
  managerName: string
  startDate: string
  endDate: string
  budgetAmount: number
  currency: string
  remark: string
}

export interface ProjectUpdateCommand {
  id: number
  projectName: string
  description: string
  departmentId: number
  departmentName: string
  managerId: number
  managerName: string
  startDate: string
  endDate: string
  budgetAmount: number
  remark: string
}

export const getProjectList = (params: {
  pageNum: number
  pageSize: number
  keyword?: string
  status?: string
  departmentId?: number
}) => {
  return http.get<CommonResult<{ list: Project[]; total: number }>>('/project-service/projects/page', { params })
}

export const getProjectById = (id: number) => {
  return http.get<CommonResult<Project>>(`/project-service/projects/${id}`)
}

export const createProject = (data: ProjectCreateCommand) => {
  return http.post<CommonResult<Project>>('/project-service/projects', data)
}

export const updateProject = (data: ProjectUpdateCommand) => {
  return http.put<CommonResult<Project>>('/project-service/projects', data)
}

export const deleteProject = (id: number) => {
  return http.delete<CommonResult<void>>(`/project-service/projects/${id}`)
}

export const updateProjectStatus = (id: number, status: string) => {
  return http.put<CommonResult<Project>>(`/project-service/projects/${id}/status`, null, { params: { status } })
}

export const updateProjectProgress = (id: number, progress: number) => {
  return http.put<CommonResult<Project>>(`/project-service/projects/${id}/progress`, null, { params: { progress } })
}